ql/src/java/org/apache/hadoop/hive/ql/metadata/client/SessionMetaStoreClientProxy.java

```diff
@@ -0,0 +1,3242 @@
+            val = (Double.valueOf(partVal1) - Double.valueOf(partVal2)) > 0 ? 1 : -1;
+          }
+          return isAsc ? val : - val;
+        }
+      }
+
+      try {
+        return Warehouse.makePartName(table.getPartitionKeys(), o1.getValues()).compareTo(
+            Warehouse.makePartName(table.getPartitionKeys(), o2.getValues()));
+      } catch (MetaException e) {
+        throw new RuntimeException(e);
+      }
+    }
+  }
+
+  @Override
+  public List<Partition> listPartitions(String catName, String dbName, String tblName, int maxParts)
+      throws TException {
+    Table table = getTempTable(dbName, tblName);
+    if (table == null) {
+      return getDelegate().listPartitions(catName, dbName, tblName, maxParts);
+    }
+    TempTable tt = getPartitionedTempTable(table);
+    return getPartitionsForMaxParts(tt.listPartitions(), maxParts);
+  }
+
+  @Override
+  public List<Partition> listPartitions(String catName, String dbName, String tblName,
+      List<String> partVals, int maxParts) throws TException {
+    Table table = getTempTable(dbName, tblName);
+    if (table == null) {
+      return getDelegate().listPartitions(catName, dbName, tblName, partVals, maxParts);
+    }
+    TempTable tt = getPartitionedTempTable(table);
+    return getPartitionsForMaxParts(tt.getPartitionsByPartitionVals(partVals), maxParts);
+  }
+
+  @Override
+  public PartitionSpecProxy listPartitionSpecs(String catName, String dbName, String tableName,
+      int maxParts) throws TException {
+    Table table = getTempTable(dbName, tableName);
+    if (table == null) {
+      return getDelegate().listPartitionSpecs(catName, dbName, tableName, maxParts);
+    }
+    TempTable tt = getPartitionedTempTable(table);
+    return getPartitionSpecProxy(table, tt.listPartitions(), maxParts);
+  }
+
+  @Override
+  public Partition getPartitionWithAuthInfo(String catName, String dbName, String tableName,
+      List<String> pvals, String userName, List<String> groupNames) throws TException {
+    Table table = getTempTable(dbName, tableName);
+    if (table == null) {
+      return getDelegate().getPartitionWithAuthInfo(catName, dbName, tableName, pvals, userName, groupNames);
+    }
+    TempTable tt = getPartitionedTempTable(table);
+    Partition partition = tt.getPartitionWithAuthInfo(pvals, userName, groupNames);
+    if (partition == null) {
+      throw new NoSuchObjectException("Partition with partition values " +
+          (pvals != null ? Arrays.toString(pvals.toArray()) : "null") +
+          " for table " + tableName + " in database " + dbName + " and for user " +
+          userName + " and group names " + (groupNames != null ? Arrays.toString(groupNames.toArray()) : "null") +
+          " is not found.");
+    }
+    return HiveMetaStoreClientUtils.deepCopy(partition);
+  }
+
+  @Override
+  public boolean dropPartition(String catName, String dbName, String tblName, List<String> partVals,
```

Comment:
I guess we have to override all dropPartition variants so that all of them handle tmp tables.